geocoding.go

```diff
@@ -169,6 +207,29 @@ func forwardGeocode(ctx context.Context, client *Client, req *ForwardGeocodeRequ
 	return &response, nil
 }
 
+// https://docs.mapbox.com/api/search/geocoding/#batch-geocoding
+func forwardGeocodeBatch(ctx context.Context, client *Client, req ForwardGeocodeBatchRequest) (*GeocodeBatchResponse, error) {
+	query := url.Values{}
```

Comment:
`url.Values` setup in f`orwardGeocode`, `reverseGeocode`, and batch geocoding functions has redundancy.

May be introduce a helper function something like this one below and use it here?
```
func prepareForwardQuery(req ForwardGeocodeRequest, apiKey string) url.Values {
    query := url.Values{}
    query.Set("q", req.SearchText)
    query.Set("access_token", apiKey)
    query.Set("autocomplete", strconv.FormatBool(req.Autocomplete))

    if !req.BBox.Min.IsZero() {
        query.Set("bbox", req.BBox.query())
    }
    if req.Country != "" {
        query.Set("country", req.Country)
    }
    if req.Language != "" {
        query.Set("language", req.Language)
    }
    if req.Limit != 0 {
        query.Set("limit", strconv.Itoa(req.Limit))
    }
    if !req.Proximity.IsZero() {
        query.Set("proximity", req.Proximity.WGS84Format())
    }
    if len(req.Types) != 0 {
        query.Set("types", req.Types.query())
    }

    return query
}

func forwardGeocode(ctx context.Context, client *Client, req *ForwardGeocodeRequest) (*GeocodeResponse, error) {
    query := prepareForwardQuery(*req, client.apiKey)

    apiResponse, err := client.get(ctx, GeocodingForwardEndpoint, query)
    if err != nil {
        return nil, fmt.Errorf("failed to get forward geocode response: %w", err)
    }

    var response GeocodeResponse
    if err := client.handleResponse(apiResponse, &response, GeocodingRateLimit); err != nil {
        return nil, fmt.Errorf("failed to handle forward geocode response: %w", err)
    }

    return &response, nil
}
```